biosppy/features/frequency.py

```diff
@@ -0,0 +1,237 @@
+    assert len(signal) > 0, 'Signal size < 1'
+
+    # ensure numpy
+    signal = np.array(signal)
+    #f_w, sp_w = st.welch_spectrum(signal, sampling_rate=FS)
+    #f_PS, sp_PS = st.power_spectrum(signal, sampling_rate=FS)
+    #f, spectrum = st.welch_spectrum(signal, size=len(signal)//2, sampling_rate=FS)
+    #f_scW, ps_scW = scipy.signal.welch(signal)
+    #f_scW = np.abs(f_scW*FS)
+    window = np.hamming(len(signal))
+    signal = signal * window
+
+    spectrum = np.abs(np.fft.fft(signal, SR))
+    
+    f = np.nan_to_num(np.array(np.fft.fftfreq(len(spectrum))))
+    spectrum = np.nan_to_num(spectrum[:len(spectrum)//2])
+    spectrum /= len(spectrum)
+    #f_2 = np.nan_to_num(np.linspace(0, FS/2, len(sp_2)))
+    f = np.abs(f[:len(f)//2]*SR)
+
+    args, names = [], []
+
+    # temporal
+    _fts = time.time_features(spectrum, SR)
+    fts_name = [str("FFT_" + i) for i in _fts.keys()]
+    fts = list(_fts[:])
+
+    args += fts
+    names += fts_name
+
+    # fundamental_frequency
+    try:
+        fundamental_frequency = f[np.argmax(spectrum)]
+    except Exception as e:
+        print(e)
+        fundamental_frequency = None
+    args += [fundamental_frequency]
+    names += ['fundamental_frequency']
+
+    ## harmonic sum
+    try:
+        if fundamental_frequency > 0:
+            harmonics = np.array([n * fundamental_frequency for n in range(2, int((SR/2)/fundamental_frequency), 1)]).astype(int)
+            sp_hrm = spectrum[np.array([np.where(f >= h)[0][0] for h in harmonics])]
+            sum_harmonics = np.sum(sp_hrm)
+        else:
+            sum_harmonics = 0
+    except Exception as e:
+        print(e)
+        sum_harmonics = None
+    args += [sum_harmonics]
+    names += ['sum_harmonics']
+
+    # spectral_roll_on
+    en_sp = spectrum**2#*(f[1]-f[0])
+    cum_en = np.cumsum(en_sp)
+
+    try:
+        norm_cm_s = cum_en/cum_en[-1]
+    except Exception as e:
+        print(e)
+        norm_cm_s =cum_en
+    try:
+        spectral_roll_on = f[np.argwhere(norm_cm_s >= 0.05)[0][0]]
+    except:
+        spectral_roll_on = None
+
+    args += [spectral_roll_on]
+    names += ['spectral_roll_on']
+
+    # spectral_roll_off
+    try:
+        spectral_roll_off = f[np.argwhere(norm_cm_s >= 0.95)[0][0]]
+    except:
+        spectral_roll_off = None
+    args += [spectral_roll_off]
+    names += ['spectral_roll_off']
+
+
+    # histogram
+    try:
+        _hist = list(np.histogram(spectrum, bins=5)[0])
+        _hist = _hist/np.sum(_hist)
+    except:
+        _hist = [None] * 5
+
+    args += [i for i in _hist]
+    names += ['spectral_hist_' + str(i) for i in range(len(_hist))]
+
+    ## bands
+    spectrum = np.nan_to_num(np.abs(np.fft.fft(signal, SR*5)))
+    f = np.nan_to_num(np.array(np.fft.fftfreq(len(spectrum))))
+
+    spectrum = np.nan_to_num(spectrum[:len(spectrum)//2])
+    spectrum /= len(spectrum)
+    f = np.abs(f[:len(f)//2]*SR)
+
+    _f = interpolate.interp1d(f, spectrum)
+    resSR = 500
```

Comment:
Change uppercase variables

```suggestion
    resampling_sr = 500
```